code/modules/admin/verbs/adminpm.dm

```diff
@@ -71,79 +71,77 @@
-		//send a warning to admins, but have a delay popup for mods
-		if(holder.rights & R_ADMIN)
-			to_chat(src, "<span class='danger'>Simultaneous PMs warning:</span> that player has been PM'd in the last [config.simultaneous_pm_warning_timeout / 10] seconds by: [C.ckey_last_pm]")
-		else
-			if(alert("That player has been PM'd in the last [config.simultaneous_pm_warning_timeout / 10] seconds by: [C.ckey_last_pm]","Simultaneous PMs warning","Continue","Cancel") == "Cancel")
-				return*/
-
+	//is this person an admin/mentor? is this person either an admin or a mentor or hold some sort of rank, if so, set them as the value. if not, set as false.
+	var/admin = check_rights(R_ADMIN|R_MOD, 0) ? src : FALSE
+	var/mentor = check_rights(R_MENTOR, 0) ? src : FALSE
+	var/ranked = (admin || mentor) ? src : FALSE
 	//get message text, limit it's length.and clean/escape html
 	if(!msg)
 		set_typing(C, TRUE)
 		msg = clean_input("Message:", "Private message to [holder ? key_name(C, FALSE) : key_name_hidden(C, FALSE)]", , src)
 		set_typing(C, FALSE)
 
-		if(!msg)
-			return
-		if(!C)
-			if(holder)
-				to_chat(src, "<span class='danger'>Error: Admin-PM: Client not found.</span>")
-			else
-				adminhelp(msg)	//admin we are replying to has vanished, adminhelp instead
-			return
+	if(handle_spam_prevention(msg, MUTE_ADMINHELP, OOC_COOLDOWN) || !msg)
+		return
 
-	if(handle_spam_prevention(msg, MUTE_ADMINHELP, OOC_COOLDOWN))
+	if(!C)
+		if(ranked)
+			to_chat(src, "<span class='danger'>Error: Private-Message: Client not found.</span>")
+		else
+			adminhelp(msg)
 		return
+	//Check for open tickets
+	var/tickets = list()
+	var/target
+	if(type == "Mentorhelp")
+		target = ranked ? C : src
+		tickets = SSmentor_tickets.checkForTicket(target)
+	else
+		target = admin ? C : src
+		tickets = SStickets.checkForTicket(target)
+
+	if(tickets)
+		for(var/datum/ticket/T in tickets)
+			T.addResponse(src, msg) // Add this response to their open tickets.
+			if(admin && (type != "Mentorhelp")) //See if we are dealing with an admin and avoid autotaking mentorhelps. Handle auto take when admin sends PM  to avoid duplicate PMing
+				if(T.staffAssigned && (T.staffAssigned != admin)) //if there is currently an admin assigned and if so, is this admin not the admin assigned to the ticket?
+					if(alert("[T.staffAssigned] is already responding, are you sure you want to continue?", ,"Yes","No") != "Yes") //ask them if they wanna continue
+						return //if they don't, then return
+				if(!T.staffAssigned) //verify that the person we are dealing with is an admin to avoid setting the assignedStaff as nobody (FALSE)
+					T.assignStaff(admin) //If nobody is assigned, we want to assign the admin to avoid duplicate PMing
 
 	//clean the message if it's not sent by a high-rank admin
 	if(!check_rights(R_SERVER|R_DEBUG,0))
 		msg = sanitize_simple(copytext(msg,1,MAX_MESSAGE_LEN))
-		if(!msg)
-			return
```

Comment:
This should probably be re-added now that I am looking at it, since it's after the sanitize

Reply:
eh I guess it could stay the way it is